content/SDKs/python/api-reference.mdx

```diff
@@ -485,6 +486,181 @@ except Exception as e:
     # Handle the error appropriately
 ```
 
+### Object Storage
+
+The Object Storage API provides a way to store and retrieve objects (files, documents, media) with support for public URL generation. It is accessed through the `context.objectstore` object.
+
+#### `async get(bucket: str, key: str) -> DataResult`
+
+Retrieves an object from the object storage.
+
+##### Parameters
+
+- `bucket`: The bucket to get the object from
+- `key`: The key of the object to get
+
+##### Return Value
+
+Returns a DataResult object that has an `exists` property to check if the object exists and a `data` property containing the object data if found.
+
+##### Example
+
+```python
+from agentuity import AgentContext
+
+# Retrieve an object from object storage with error handling
+try:
+    result = await context.objectstore.get("user-uploads", "profile-123.jpg")
+    if result.exists:
+        # Access data using the Data interface methods
+        image_data = await result.data.binary()
+        context.logger.info(f"Image size: {len(image_data)} bytes")
+    else:
+        context.logger.info("Image not found")
+except Exception as e:
+    context.logger.error(f"Failed to retrieve object: {str(e)}")
+    # Handle the error appropriately
+```
+
+#### `async put(bucket: str, key: str, data: DataLike, params: Optional[ObjectStorePutParams] = None) -> None`
+
+Stores an object in the object storage.
+
+##### Parameters
+
+- `bucket`: The bucket to put the object into
+- `key`: The key of the object to put
+- `data`: The data to store (can be string, bytes, or other DataLike types)
+- `params` (optional): Additional parameters for the object. The following properties map to canonical HTTP headers: `Content-Type`, `Content-Encoding`, `Cache-Control`, `Content-Disposition`, and `Content-Language`. All other properties are added as generic metadata to the underlying storage provider (if supported). The `Content-Type` parameter is optional - if not provided, the content type will be automatically detected based on the data type (e.g., objects will be set to `application/json`). If the content type cannot be determined, it defaults to `application/octet-stream`.
```

Comment:
ok, i was incorrect.

for params this is an optional object of type ObjectStorePutParams. it has the following optional properties:

- content_type
- content_encoding
- cache_control
- content_disposition
- content_language

it also has an optional property `metadata` for storing arbitrary metadata associated with the object. this metadata will not be set of any HTTP results for the object when it is streamed.